src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java

```diff
@@ -0,0 +1,178 @@
+package org.geoserver.gwc.config;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geoserver.data.util.IOUtils;
+import org.geoserver.platform.resource.Files;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.util.Filter;
+import org.geowebcache.config.ConfigurationException;
+import org.geowebcache.config.ConfigurationResourceProvider;
+
+public class GeoserverXMLResourceProvider implements ConfigurationResourceProvider {
+
+    private static Log log = LogFactory.getLog(org.geowebcache.config.XMLFileResourceProvider.class);
+    
+    public static final String DEFAULT_CONFIGURATION_DIR_NAME = "gwc";
+    
+    /**
+     * Location of the configuration file
+     */
+    private final Resource configDirectory;
+
+    /**
+     * Name of the configuration file
+     */
+    private final String configFileName;
+    
+    private String templateLocation;
+    
+    public GeoserverXMLResourceProvider(final String configFileDirectory,
+            final String configFileName,
+            final ResourceStore resourceStore) throws ConfigurationException {
+        
+        this.configFileName = configFileName;
+        
+        if(configFileDirectory!=null) {
+            // Use the given path
+            if (configFileDirectory.startsWith("/") || configFileDirectory.contains(":\\")
```

Comment:
@NielsCharlier there must be a better way to do this. How about this?
'''
if ((new File(configFileDirectory)).isAbsolute()) {
'''